### ISSUE TITLE
Use backticks instead of italics for literal types.

### ISSUE THREAD

**Issue by KillyMXI** [tgreyuk/typedoc-plugin-markdown#212]
... and escape values properly.

Just escaping can't help with spaces, just backticks leave vertical bar issue.

Input like this

```typescript
export type foo = {
  bar: ' ' | '|' | '*'
}
```

produces a table like this

```md
Name | Type |
:------ | :------ |
`bar` | * * \| *|* \| *** |
```

which can't be parsed and rendered properly.

Don't let arbitrary code spill over! (Not to mention it's currently not possible to distinguish literal types from non-literal and that has a chance to create a confusion.)

What I expect to see:

```md
| Name | Type
| :--- | :---
| `bar` | ` ` \| `\|` \| `*`
```

| Name | Type
| :--- | :---
| `bar` | ` ` \| `\|` \| `*`

*(Offtopic note: Table bars at the beginning of the line is another preference of mine - they make raw markdown more readable. I can make a PR if you like.)*

**Comment by tgreyuk** [comment]
@KillyMXI thanks for this.

* Agreed backticks make sense for type literals. I have actually implemented double backticks just to cover any values that contain backticks in the value itself.
* I thought it also makes sense to follow the HTML theme which wraps the value in JSON.stringyfy and quotes string values.
* Have also added table bars to beggining of table lines as per your suggestion.

So after the code change we will have the following:

Markdown:

```
| Name | Type |
| :------ | :------ |
| `bar` | ``" "`` \| ``"\|"`` \| ``"*"`` |
```

Preview:

![image](https://user-images.githubusercontent.com/11680870/115120123-b16b0a80-9fa3-11eb-8144-aae2f29867db.png)

Let me know if you think this works?

**Comment by KillyMXI** [comment]
This is great, thank you!

Yeah, I should've mentioned about quotes. I wasn't sure without a closer look whether they can be reinserted safely (without accidentally quoting number and booleans).

Compared to the first bar in a table row, I find the last bar rather unnecessary. But that might be debatable and fine by me either way.

So yeah, I'm happy about this solution.

**Comment by tgreyuk** [comment]
ok cool - update available in typedoc-plugin-markdown@3.7.0

let me know if you have any issues..

**Comment by KillyMXI** [comment]
It works great now, thank you once again.

One weird but unrelated thing - a table of function parameters has extra spaces after a description. Is there a reason for that?

```md
| Name | Type | Description |
| :------ | :------ | :------ |
| `str` | *string* | Three spaces after me.    |
```

Minor think, just bothers my OCD.

**Comment by tgreyuk** [comment]
No good spot - i will tidy that up.

**Comment by tgreyuk** [comment]
Fixed the extraneous whitespace in 3.7.1. Closing this issue. Feel free to raise new ticket if you spot anything else. thanks.

**Comment by KillyMXI** [comment]
Awesome.
The document that run me into this issue is [now public](https://github.com/mxxii/parseley/blob/main/docs/modules/ast.md), in case you're curious.
